pkg/tpkg/registry.go

```diff
@@ -337,97 +349,105 @@ func (gr *gitRegistry) Describe() string {
 	return fmt.Sprintf("%s: %s", gr.name, gr.url)
 }
 
-func (gr *gitRegistry) Load(ctx context.Context, sync bool, cache Cache, ui UI) error {
-	if sync {
-		p := gr.path
-		if gr.path == "" {
-			p = cache.PreferredRegistryPath(gr.url)
-		}
+func (gr *gitRegistry) withFileLock(ctx context.Context, cache Cache, f func(path string) error) error {
+	p := gr.path
```

Comment:
This code was previously in the 'Load' function.
Only change is the `return f(p)` in the last line.